homeassistant/components/decora_wifi/light.py

```diff
@@ -109,37 +72,44 @@ def is_on(self):
         """Return true if switch is on."""
         return self._switch.power == "ON"
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs):
         """Instruct the switch to turn on & adjust brightness."""
         attribs = {"power": "ON"}
 
         if ATTR_BRIGHTNESS in kwargs:
-            min_level = self._switch.data.get("minLevel", 0)
-            max_level = self._switch.data.get("maxLevel", 100)
-            brightness = int(kwargs[ATTR_BRIGHTNESS] * max_level / 255)
-            brightness = max(brightness, min_level)
+            maxlevel = self._switch.data.get("maxLevel", 100)
+            minlevel = self._switch.data.get("minLevel", 100)
+            brightness = int(kwargs[ATTR_BRIGHTNESS] * maxlevel / 255)
+            brightness = max(brightness, minlevel)
             attribs["brightness"] = brightness
 
         if ATTR_TRANSITION in kwargs:
             transition = int(kwargs[ATTR_TRANSITION])
             attribs["fadeOnTime"] = attribs["fadeOffTime"] = transition
 
-        try:
+        def tryupdate():
             self._switch.update_attributes(attribs)
+
+        try:
+            await self.hass.async_add_executor_job(tryupdate)
         except ValueError:
             _LOGGER.error("Failed to turn on myLeviton switch")
 
-    def turn_off(self, **kwargs):
+    async def async_turn_off(self, **kwargs):
         """Instruct the switch to turn off."""
         attribs = {"power": "OFF"}
-        try:
+
+        def tryupdate():
             self._switch.update_attributes(attribs)
+
+        try:
+            self._switch.update_attributes(tryupdate)
         except ValueError:
             _LOGGER.error("Failed to turn off myLeviton switch")
 
-    def update(self):
+    async def async_update(self):
         """Fetch new state data for this switch."""
         try:
-            self._switch.refresh()
+            await self.hass.async_add_executor_job(self._switch.refresh)
         except ValueError:
             _LOGGER.error("Failed to update myLeviton switch data")
```

Comment:
Based on the response above, this is an example of unneeded async wrapping a sync API. This could just have been `def update(self):`, there is no need to wrap this.

Reply:
Understood. Will work on it this evening.